### ISSUE TITLE
An attempt was made to load a program with an incorrect format.

### ISSUE THREAD

**Issue by Sgierek106** [ErikEJ/EFCorePowerTools#1032]
I installed the latest version of the extension this morning, and now whenever I attempt to Edit or Refresh my efpt.config.json, it throws the following exception:
```
System.Exception: Table list error: 
Unhandled exception. System.BadImageFormatException: Could not load file or assembly '%USERPROFILE%\AppData\Local\Temp\2\efreveng5.2.5.691.0\efreveng50.dll'. An attempt was made to load a program with an incorrect format.
File name: '%USERPROFILE%\AppData\Local\Temp\2\efreveng5.2.5.691.0\efreveng50.dll'

   at List<TableModel> EFCorePowerTools.Handlers.ReverseEngineer.ResultDeserializer.BuildTableResult(string output)
   at async Task<List<TableModel>> EFCorePowerTools.Handlers.ReverseEngineer.EfRevEngLauncher.GetTablesInternalAsync(string arguments)
   at async Task<List<TableModel>> EFCorePowerTools.Handlers.ReverseEngineer.EfRevEngLauncher.GetTablesAsync(string connectionString, DatabaseType databaseType, SchemaInfo[] schemas)
   at async Task<List<TableModel>> EFCorePowerTools.Handlers.ReverseEngineer.TableListBuilder.GetTableDefinitionsAsync(CodeGenerationMode codeGenerationMode)
   at async Task<List<TableModel>> EFCorePowerTools.Handlers.ReverseEngineer.ReverseEngineerHandler.GetTablesAsync(DatabaseConnectionModel dbInfo, CodeGenerationMode codeGenerationMode, SchemaInfo[] schemas)
   at async Task<bool> EFCorePowerTools.Handlers.ReverseEngineer.ReverseEngineerHandler.LoadDataBaseObjectsAsync(ReverseEngineerOptions options, DatabaseConnectionModel dbInfo, Tuple<List<Schema>, string> namingOptionsAndPath)
   at async Task EFCorePowerTools.Handlers.ReverseEngineer.ReverseEngineerHandler.ReverseEngineerCodeFirstAsync(Project project, string optionsPath, bool onlyGenerate)
```
### Steps to reproduce

1. Right-click efpt.config.json and select "EF Core Power Tools - Refresh".
2. The error will be displayed in the Output window under "Extensions".

### Further technical details
EF Core Power Tools version: 2.5.691.0

Database engine: SQL Server

Visual Studio version: Visual Studio 2019 Professional 16.10.2

**Comment by ErikEJ** [comment]
Can you launch reverse engineering from the project menu?

Do you have the 3.1 x64 runtime installed?

(dotnet --list-runtimes)

**Comment by ErikEJ** [comment]
I am not able to repro...

**Comment by ErikEJ** [comment]
I will try to repro

**Comment by Sgierek106** [comment]
Hello!

While I was gathering the information to answer your questions, I discovered the root cause of my issue!

When I opened a command prompt to run the "dotnet --list-runtimes" command, I noticed that it only showed me the 32-bit versions (i.e., those under "C:\Program Files (x86)")

On a hunch, I opened up my system Environment Variables to look at the PATH variable, and lo and behold, the "C:\Program Files (x86)\dotnet" path was listed before the "C:\Program Files\dotnet" path.  So, Visual Studio was trying to use the 32-bit version of the 3.1 runtime instead of the x64.

The fix was simply to move the x64 path above the 32-bit path and then restart Visual Studio:

![dotnet-system-vars](https://user-images.githubusercontent.com/8323762/123122309-c11b2b00-d3fa-11eb-85df-f7380465ce48.png)

Thank you for the quick replies, and for making such an awesome extension!  I would not be using EF Core if this didn't exist.
